lib/membrane/video_compositor/scene/object/layout.ex

```diff
@@ -37,4 +54,80 @@ defmodule Membrane.VideoCompositor.Scene.Object.Layout do
           :resolution => output_resolution(),
           optional(any()) => any()
         }
+
+  defmodule RustlerFriendly do
+    @moduledoc false
+    # A rustler-friendly version of the Layout, prepared for rust serialization
+    alias Membrane.VideoCompositor.Scene.Object.Layout
+    alias Membrane.VideoCompositor.Scene.Object.RustlerFriendly, as: RFObject
+    alias Membrane.VideoCompositor.Scene.Resolution
+
+    @typedoc """
+    A rustler-friendly version of the internal name
+    """
+    @type internal_name :: RFObject.name()
+
+    @typedoc """
+    A rustler-friendly version of the inputs map. Keep in mind that this maps
+    __internal names__ => __scene object names__.
+    """
+    @type inputs :: %{internal_name() => RFObject.name()}
+    @type output_resolution :: {:resolution, Resolution.t()} | {:name, Object.name()}
+
+    @type t :: %__MODULE__{
+            :inputs => inputs(),
+            :resolution => output_resolution(),
+            # unsure about calling this `implementation`.
+            :implementation => Layout.rust_representation()
+          }
+
+    @enforce_keys [:inputs, :resolution, :implementation]
+    defstruct @enforce_keys
+  end
+
+  @typedoc """
+  This documentation is temporary, since something needs to be here and it
+  is currently unknown what this will look like.
+
+  in a more 'final' product this should be some kind of a layout identifier.
+  I thought of making this a UUID that would correspond to an implementation
+  on the rust side, but layout names would work fine too.
+  """
+  @type rust_representation :: integer()
+
+  @doc """
+  A callback used for encoding the static layout data into a rust-based representation.
+  This should be implemented in user-defined layouts. This function is responsible for encoding
+  the optional fields and all other parts of the layout into the user-defined rust struct.
+  We don't know yet how exactly this system is going to work, so this is just a placeholder
+  for now.
+  """
+  @callback encode(t()) :: rust_representation()
+
+  @doc false
+  # Encode the layout to a Layout.RustlerFriendly in order to prepare it for
+  # the rust conversion.
+  @spec encode(t()) :: RustlerFriendly.t()
+  def encode(layout = %module{inputs: inputs, resolution: resolution}) do
+    rust_representation = module.encode(layout)
```

Comment:
As we discussed in person, this is rather unnecessary at this stage. perhaps it will be beneficial in the future, and we can implement it then. Right now it is rather pointless.